convert2rhel/main.py

```diff
@@ -79,40 +77,39 @@ def main():
     toolopts.CLI()
     try:
         process_phase = ConversionPhase.POST_CLI
+        perform_boilerplate()
 
-        # license agreement
-        loggerinst.task("Prepare: Show Red Hat software EULA")
-        show_eula()
-
-        loggerinst.task("Prepare: Inform about telemetry")
-        breadcrumbs.breadcrumbs.print_data_collection()
-
-        # gather system information
-        loggerinst.task("Prepare: Gather system information")
-        systeminfo.system_info.resolve_system_info()
-        systeminfo.system_info.print_system_information()
-
-        breadcrumbs.breadcrumbs.collect_early_data()
-
-        loggerinst.task("Prepare: Clear YUM/DNF version locks")
-        pkghandler.clear_versionlock()
+        gather_system_info()
+        prepare_system()
 
-        loggerinst.task("Prepare: Clean yum cache metadata")
-        pkgmanager.clean_yum_metadata()
-
-        # check the system prior the conversion (possible inhibit)
-        checks.perform_system_checks()
-
-        # backup system release file before starting conversion process
-        loggerinst.task("Prepare: Backup System")
-        redhatrelease.system_release_file.backup()
-        redhatrelease.os_release_file.backup()
-        repo.backup_yum_repos()
-        repo.backup_varsdir()
-
-        # begin conversion process
+        # Note: set pre_conversion_results before changing to the next phase so
+        # we don't fail in case rollback is triggered during
+        # actions.run_actions() (either from a bug or from the user hitting
+        # Ctrl-C)
+        pre_conversion_results = None
         process_phase = ConversionPhase.PRE_PONR_CHANGES
-        pre_ponr_conversion()
+        pre_conversion_results = actions.run_actions()
+
+        experimental_analysis = bool(os.getenv("CONVERT2RHEL_EXPERIMENTAL_ANALYSIS", None))
+        if experimental_analysis:
+            process_phase = ConversionPhase.ANALYZE_EXIT
+            raise _AnalyzeExit()
+
+        pre_conversion_failures = actions.find_actions_of_severity(pre_conversion_results, "SKIP")
+        if pre_conversion_failures:
+            # The report will be handled in the error handler, after rollback.
+            loggerinst.critical("Conversion failed.")
+
+        #
+        # Ready to convert
+        #
```

Comment:
I put this in because it felt like this was a major transition from pre-PONR work to knowing we can run a conversion.... But maybe it isn't needed (or if it is needed, we'd be better served by splitting this into two functions).

I'll remove it